manager/scheduler/filter.go

```diff
@@ -145,8 +145,18 @@ func (f *PluginFilter) Check(n *NodeInfo) bool {
 
 	// Check if all network plugins required by task are installed on node
 	for _, tn := range f.t.Networks {
-		if tn.Network != nil && tn.Network.DriverState != nil && tn.Network.DriverState.Name != "" {
-			if !f.pluginExistsOnNode("Network", tn.Network.DriverState.Name, nodePlugins) {
+		if tn.Network == nil {
+			continue
+		}
+		if tn.Network.Spec.GetCNM() == nil {
+			continue // Non-CNM Networks have no plugin requirements
+		}
+		cnmState := tn.Network.GetCNM()
+		if cnmState == nil {
+			panic("CNM network with no CNM state")
```

Comment:
As I mentioned above, I'd like to explore the idea of falling back on the legacy fields when `CNMSpec` does not exist. But in the case of a mismatch between the types inside "spec" and "state", I think it's better not to panic. The scheduler is a tricky case because there aren't really opportunities to bubble errors up from here. I see three options:

1) Silently ignore these mismatches in the scheduler, and assume they are logged somewhere else, such as the allocator. Either disable the filter or have it return false for all nodes (so the affected task will not be scheduled).
2) Log a message, rate-limited with the `golang.org/x/time/rate` package
3) Change the filter design so it can bubble up errors. These errors would be put into `Task.Status.Err` and would be visible when inspecting a task (though discoverability of this feature is quite poor).

Exposing errors from scheduler filters also came up in #2090.